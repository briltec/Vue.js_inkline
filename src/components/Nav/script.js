import ClassesProviderMixin from 'inkline/mixins/components/providers/ClassesProviderMixin';

import SizePropertyMixin from 'inkline/mixins/components/properties/SizePropertyMixin';

export default {
    name: 'INavbarBrand',
    mixins: [
        ClassesProviderMixin,

        SizePropertyMixin,
    ],
    props: {
        tabs: {
            type: Boolean,
            default: false
        },
        vertical: {
            type: Boolean,
            default: false
        }
    },
    created () {
        if (this.classesProvider) {
            this.classesProvider['root'].push(() => ({
                '-tabs': this.tabs,
                '-vertical': this.vertical
            }));
        }
    }
};