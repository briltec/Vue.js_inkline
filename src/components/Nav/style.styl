/**
 * Nav
 */

@import '../../css/config'
@import '../../css/mixins'
@import 'css/_variables'
@import 'css/_sizes'

.nav
    display: flex
    flex-wrap: wrap
    padding-left: 0
    margin-bottom: 0
    list-style: none

    &.-vertical
        flex-direction: column

    @extend .nav.-md

