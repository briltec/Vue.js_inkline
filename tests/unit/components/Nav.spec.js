import { mount } from '@vue/test-utils';
import Nav from 'inkline/components/Nav';

describe('Components', () => {
    describe('Nav', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = mount(Nav);
        });

        test('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
