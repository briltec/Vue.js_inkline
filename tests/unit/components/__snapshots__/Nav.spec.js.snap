// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Nav should render correctly 1`] = `
<div class="alert">
  <!---->
  <div class="content"></div>
  <!---->
</div>
`;
